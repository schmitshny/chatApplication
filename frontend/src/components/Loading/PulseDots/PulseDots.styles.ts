import styled, { keyframes } from 'styled-components';

const pulse = keyframes`
  0% {
    transform: scale(1);
    opacity: 1;
  }
  50% {
    transform: scale(1.5);
    opacity: 0.7;
  }
  100% {
    transform: scale(1);
    opacity: 1;
  }
`;

export const PulseDot = styled.div<{ $delay?: string }>`
  width: 10px;
  height: 10px;
  background-color: #007bff;
  border-radius: 50%;
  margin: 0 5px;
  animation: ${pulse} 1.5s infinite ease-in-out;
  animation-delay: ${({ $delay = '0s' }) => $delay};
`;

export const LoaderWrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  height: 100vh;
  width: 100%;
`;
