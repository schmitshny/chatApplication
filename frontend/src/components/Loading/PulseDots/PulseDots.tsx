import { LoaderWrapper, PulseDot } from './PulseDots.styles';

export const PulseDots = () => (
  <LoaderWrapper data-testid="pulse-dots">
    <PulseDot $delay="0s" />
    <PulseDot $delay="0.5s" />
    <PulseDot $delay="1s" />
  </LoaderWrapper>
);
