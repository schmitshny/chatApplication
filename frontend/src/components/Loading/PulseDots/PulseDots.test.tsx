import { describe, test, expect } from 'vitest';
import { render } from '@testing-library/react';
import { PulseDots } from './PulseDots';

describe('PulseDots', () => {
  test('renders three PulseDot components', () => {
    const { container, getByTestId } = render(<PulseDots />);

    const loaderWrapper = getByTestId('pulse-dots');

    expect(loaderWrapper.children).toHaveLength(3);
    expect(container).toMatchSnapshot();
  });
});
