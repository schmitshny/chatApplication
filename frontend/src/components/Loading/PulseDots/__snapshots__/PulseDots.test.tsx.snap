// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PulseDots > renders three PulseDot components 1`] = `
<div>
  <div
    class="sc-gsnScm fhqUow"
    data-testid="pulse-dots"
  >
    <div
      class="sc-bczSft kcJqPc"
    />
    <div
      class="sc-bczSft lfjsYf"
    />
    <div
      class="sc-bczSft kcJquh"
    />
  </div>
</div>
`;
